core/src/main/java/software/amazon/awssdk/http/ExecutionContext.java

```diff
@@ -122,6 +123,7 @@ public void setCredentialsProvider(AwsCredentialsProvider credentialsProvider) {
         this.credentialsProvider = credentialsProvider;
     }
 
+    @ReviewBeforeRelease("Not used. Should it be?")
```

Comment:
Looks like it can be, we get it from the RequestExecutionContext.

https://github.com/aws/aws-sdk-java-v2/blob/7195fdb88f25885a9c2621b70f4304fbd839ec38/core/src/main/java/software/amazon/awssdk/http/pipeline/stages/MakeHttpRequestStage.java#L64